onebusaway-android/src/main/java/com/joulespersecond/oba/ObaAnalytics.java

```diff
@@ -129,87 +131,88 @@ public static void reportEventWithCategory(String category, String action, Strin
-                stopDistance = ObaStopDistance.DISTANCE_1;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_2.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_2;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_3.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_3;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_4.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_4;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_5.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_5;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_6.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_6;
-            } else if (distanceInMeters < ObaStopDistance.DISTANCE_7.getDistanceInMeters()) {
-                stopDistance = ObaStopDistance.DISTANCE_7;
-            } else {
-                stopDistance = ObaStopDistance.DISTANCE_8;
+        if (isAnalyticsActive()) {
+            if (myLocation == null) {
+                return;
+            }
+            if (myLocation.getAccuracy() < LOCATION_ACCURACY_THRESHOLD) {
+                float distanceInMeters = myLocation.distanceTo(stopLocation);
+                ObaStopDistance stopDistance = null;
+                String obaRegionName = getObaRegionName();
+
+                if (distanceInMeters < ObaStopDistance.DISTANCE_1.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_1;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_2.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_2;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_3.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_3;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_4.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_4;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_5.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_5;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_6.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_6;
+                } else if (distanceInMeters < ObaStopDistance.DISTANCE_7.getDistanceInMeters()) {
+                    stopDistance = ObaStopDistance.DISTANCE_7;
+                } else {
+                    stopDistance = ObaStopDistance.DISTANCE_8;
+                }
+
+                Tracker tracker = Application.get().getTracker(Application.TrackerName.APP_TRACKER);
+                Tracker tracker2 = Application.get().getTracker(Application.TrackerName.GLOBAL_TRACKER);
+
+                tracker.send(new HitBuilders.EventBuilder()
```

Comment:
@bbodenmiller Yes I agree, I would be better approach to reduce redundancy.
I will update the pr. @barbeau fyi,